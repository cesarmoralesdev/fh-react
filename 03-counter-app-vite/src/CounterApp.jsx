import { useState } from 'react';
import { z } from 'zod';


const CounterApp = ({ value }) => {
    const [counter, setCounter] = useState(value);

    //Se puede poner la funcion fuera del componente (siempre y cuando no necesite un parametro del componete), pero solo funcionaria la primera vez que carga el componente, si se actualiza se perderia la referencia a la funcion
    const handleAdd = () => {
        //En versiones anteriores de React, si habia varios setXXX, por cada uno hacia un render, ahora todos los setXXX  se ejecutan y cuando terminan recien se hace el render
        setCounter(counter + 1);
    }

    const handleSubtract = () => setCounter(counter - 1);
    const handleReset = () => setCounter(value);

    const schema = z.object({ value: z.number() });

    const validatedValue = schema.parse({ value });

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <button onClick={ handleAdd }>+</button>
            <button onClick={ handleSubtract }>-</button>
            <button aria-label='btn-reset' onClick={ handleReset }>Reset</button>
        </>
    );
}
export default CounterApp;