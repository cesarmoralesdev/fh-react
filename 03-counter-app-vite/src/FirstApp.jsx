import { Fragment } from 'react'
import { z } from "zod";

export const FirstApp = (props) => {

    // Define el esquema de Zod
    const schema = z.object({
        title: z.string().default("No hay titulo"),
        subtitle: z.string().optional(), // Agregué subtitle al esquema si lo usas
    });

    // Parsea los props y obtén el objeto validado con los defaults aplicados
    const validatedProps = schema.parse(props);

    // Desestructura las propiedades del objeto validado
    const { title, subtitle } = validatedProps;

    return (
        <>
            <h1>{title}</h1>
            <p>{subtitle}</p>
        </>
    )
}

