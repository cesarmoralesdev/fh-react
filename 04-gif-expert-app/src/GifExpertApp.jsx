import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {

    // 
    //
    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball']);

    const onAddCategory = (value) => {
        setCategories([
            ...categories,
            value
        ]);
        //Forma 2: Con esta forma dentro del setState tenemos el valor de estado
        // setCategories(cat => [
        //     ...cat,
        //     value
        // ]);
    }

    return (
        <>
            {/* titulo */}
            <h3>GifExpertApp</h3>

            {/* input */}
            <AddCategory 
                // setCategories={setCategories}
                onNewCategory = { value => onAddCategory(value)}
            />

            {/* listado de Gif */}
            <ol>
                {categories.map(category => {
                    return <li key={category}>{category}</li>
                })}
            </ol>
            {/* Gif Item */}


        </>
    )
}
