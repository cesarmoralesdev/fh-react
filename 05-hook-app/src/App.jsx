import { useState } from 'react'
import './App.css'
// import { CounterApp } from './01-useState/CounterApp'
// import { CounterWithCustomHooks } from './01-useState/CounterWithCustomHooks'
import { SimpleFrom } from './02-useEffect/SimpleFrom'

export const App = () => {
  return (
    <>
      <SimpleFrom />
    </>
  )
}

export default App
