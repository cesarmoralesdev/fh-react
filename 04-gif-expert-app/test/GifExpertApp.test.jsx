import { render, screen, fireEvent } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";
describe('Pruebas en <GifExpertApp />', () => {
    test('debe de probar Snapshot', () => {
        const { container } = render(<GifExpertApp />);
        // screen.debug();
        expect(container).toMatchSnapshot();
    });
    test('debe de retornar el estado inicial', () => {
        const { container } = render(<GifExpertApp />);
        // screen.debug();
        const input = screen.getByRole('textbox');
        //fireEvent es para disparar eventos en el componente
        fireEvent.input(input, { target: { value: 'Naruto' } });
        expect(input.value).toBe('Naruto');
    });
    
})