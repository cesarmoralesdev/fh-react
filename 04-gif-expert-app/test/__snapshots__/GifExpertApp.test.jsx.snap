// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <GifExpertApp /> debe de probar Snapshot 1`] = `
<div>
  <h3>
    GifExpertApp - CÃ©sar Morales
  </h3>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
